### PR TITLE
[Core] Add rule to check for forgotten console.logs

### PR DESCRIPTION
**Pull request by alisterdev**
- [x] Add ESLint rule to check for forgotten console.logs
- [x] Remove console logs from already linted files

>Note: `console.info`, `console.warn`, `console.error` are still allowed but should only be used when appropriate

### CODE REVIEW

File: modules/genomic_browser/js/FileUploadModal.js [hunk 1]
```diff
@@ -10,19 +12,19 @@ var GenomicFileUploadModal = React.createClass({
     baseURL: React.PropTypes.string.isRequired
   },
 
-  getInitialState: function () {
+  getInitialState: function getInitialState() {
```
Comment (xlecours) [inline modules/genomic_browser/js/FileUploadModal.js]: Function name should be removed

Comment (xlecours) [inline modules/genomic_browser/js/FileUploadModal.js]: Did we agreed on that standard? I have a lot of subjective comments on that.

Comment (alisterdev) [inline modules/genomic_browser/js/FileUploadModal.js]: I think we already raised that question, and I found a decent explanation for such behaviour here: http://stackoverflow.com/a/35857404

Comment (alisterdev) [inline modules/genomic_browser/js/FileUploadModal.js]: @xlecours What is your issues with that?

**Pull request merged**